src/tests/plans/D3MPlanBase.t.sol

```diff
@@ -16,117 +16,38 @@
-        emit Rely(usr);
-    }
-    function deny(address usr) external auth {
-        wards[usr] = 0;
-        emit Deny(usr);
-    }
-    modifier auth {
-        require(wards[msg.sender] == 1, "D3MPlanBase/not-authorized");
-        _;
-    }
-
-    // --- Events ---
-    event Rely(address indexed usr);
-    event Deny(address indexed usr);
-
-    constructor(address dai_) {
-        dai = dai_;
+    ID3MPlan private plan;
 
-        wards[msg.sender] = 1;
-        emit Rely(msg.sender);
+    function baseInit(ID3MPlan _plan, string memory _contractName) internal {
+        plan = _plan;
+        contractName = _contractName;
     }
 
-    function getTargetAssets(uint256 currentAssets) external override pure returns(uint256) {
-        return currentAssets;
+    function test_auth() public {
+        checkAuth(address(plan), contractName);
     }
 
-    function active() external override pure returns(bool) {
-        return true;
-    }
-
-    function disable() external override {}
-}
-
-contract D3MPlanBaseTest is DssTest {
-    string contractName;
-
-    DaiLike dai;
-
-    address d3mTestPlan;
-
-    function setUp() public virtual {
-
-        contractName = "D3MPlanBase";
-
-        dai = DaiLike(address(123));
+    function test_auth_modifiers() public virtual {
+        WardsAbstract(address(plan)).deny(address(this));
 
-        d3mTestPlan = address(new D3MPlanBase(address(dai)));
+        checkModifier(address(plan), string(abi.encodePacked(contractName, "/not-authorized")), [
+            ID3MPlan.disable.selector
+        ]);
     }
 
-    function test_sets_creator_as_ward() public {
-        assertEq(D3MPlanBase(d3mTestPlan).wards(address(this)), 1);
+    function test_disable_makes_inactive() public virtual {
+        assertEq(plan.active(), true);
+        plan.disable();
+        assertEq(plan.active(), false);
     }
 
-    function test_can_rely() public {
-        assertEq(D3MPlanBase(d3mTestPlan).wards(address(123)), 0);
-
-        D3MPlanBase(d3mTestPlan).rely(address(123));
-
-        assertEq(D3MPlanBase(d3mTestPlan).wards(address(123)), 1);
-    }
-
-    function test_can_deny() public {
-        assertEq(D3MPlanBase(d3mTestPlan).wards(address(this)), 1);
-
-        D3MPlanBase(d3mTestPlan).deny(address(this));
-
-        assertEq(D3MPlanBase(d3mTestPlan).wards(address(this)), 0);
-    }
-
-    function test_cannot_rely_without_auth() public {
-        assertEq(D3MPlanBase(d3mTestPlan).wards(address(this)), 1);
-
-        D3MPlanBase(d3mTestPlan).deny(address(this));
-        assertRevert(d3mTestPlan, abi.encodeWithSignature("rely(address)", address(this)), string(abi.encodePacked(contractName, "/not-authorized")));
-    }
-
-    function test_implements_getTargetAssets() public virtual {
-        uint256 result = D3MPlanBase(d3mTestPlan).getTargetAssets(2);
-
-        assertEq(result, 2);
-    }
-
-    function test_implements_active() public view {
-        D3MPlanBase(d3mTestPlan).active();
-    }
```

Comment:
Shouldn't these two be part of the base? The `hub` requires them, a plan that doesn't implement them is wrong.

Reply:
That's enforced by inheriting the interface though.

Reply:
Hmmm fair, I'd think we could still leave them, but I guess we are protected anyway so not strong opinion.